Global keyboard shortcut functionality

ren-paz opened the pull request:
extracted ipcHandler to seperate function

now return created browser windows from createWindow function

Types for upcoming keyboard global shortcuts

ren-paz committed:
extracted ipcHandler to seperate function

Now return created browser windows from createWindow function

Types for upcoming keyboard global shortcuts

ren-paz committed:
more tidying up

ren-paz committed:
fix app not launching

added validation function

ren-paz committed:
Merge branch 'main' into shortcut

ren-paz committed:
fixed validation function

ren-paz committed:
added support for hotkeys and changing hotkeys

RealLacuni commented:
need to test the hotkey functionality, currently just prints to backend so that key strokes can be manually tested on windows + macOS for different combinations.

ren-paz committed:
fix circle to use all values from preferences

ren-paz committed:
display current hotkey in home menu

remove some log statements

ren-paz committed:
Merge branch 'main' of https://github.com/RealLacuni/overlay-x into shortcut

ren-paz committed:
circle changes

ren-paz committed:
fixed  bug with overlay not updating changes

refactored preload to include a separate api for Overlay window

Overlay window runs in a separate react app instance so message passing is needed

ren-paz committed:
removed app.tsx finally

fixed changing and updating hotkeys, however still buggy behavior

ren-paz committed:
added alert notification to use for setting confirmation

ren-paz committed:
seperated app bar from rest of app by fixing on top

fixed scrollbar positioning

ren-paz committed:
added rough draft of alert to save button

ren-paz committed:
button styling changes

Added invalid hotkey alert to hotkey inputs

Formatting changes for sliders.

Added max/fullscreen to the thickness input that will ensure full coverage of the screen if slider is at the maximum value

ren-paz committed:
fixed max value calculation for circle thickness

styling updates
